scipy/optimize/optimize.py

```diff
@@ -1465,7 +1465,8 @@ def _minimize_newtoncg(fun, x0, args=(), jac=None, hess=None, hessp=None,
             A = fhess(*(xk,) + args)
             hcalls = hcalls + 1
 
-        while numpy.add.reduce(numpy.abs(ri)) > termcond:
+        k2 = 0
+        while (numpy.add.reduce(numpy.abs(ri)) > termcond) and (k2 < maxiter):
```

Comment:
So, yes I think it's correct to limit it to `n` iterations. If it does `n` iterations, then check the tolerance and if it is yet not satisfied, report that the CG iterations didn't converge. (Alternatively run `n+1` iterations and if `n+1` is done, then the CG iterations didn't converge).

The message probably could be: "CG iteration didn't converge, the Hessian is not positive definite." (Sorry, forgot to mention Hessian in the initial suggestion).